### PR TITLE
Promote duck to v1

### LINKED ISSUE #1422: Promote apis/duck and duck to v1
**Problem**
Help with #1409, #1410, #1411, #1412, #1413.
Need to promote duck first as the source APIs needs the duck types. 

**[Persona:](https://github.com/knative/eventing/blob/master/docs/personas.md)** 
Which persona is this feature for?

**Exit Criteria**
pkg/apis/duck and pkg/duck are promoted to v1.

**Time Estimate (optional):** 
How many developer-days do you think this may take to resolve?

**Additional context (optional)**
Add any other context about the feature request here.

cc @nachocano @capri-xiyue

### PR DESCRIPTION
**Pull request by danyinggu**
Fixes #1422 

## Proposed Changes

- 🎁 Promote duck to v1 as Source needs this change to promote to v1.
- It doesn't change any code to use the new type.
- All code is copied from duck v1beta1.

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

**Comment by danyinggu** [comment]
@googlebot

File: pkg/apis/duck/v1/identity_types.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+// IdentityStatus inherits duck/v1 Status and adds a ServiceAccountName.
+type IdentityStatus struct {
+	// Inherits duck/v1 Status,, which currently provides:
+	// * ObservedGeneration - the 'Generation' of the Service that was last processed by the controller.
+	// * Conditions - the latest available observations of a resource's current state.
+	duckv1.Status `json:",inline"`
+	// ServiceAccountName is the k8s service account associated with Google service account.
+	ServiceAccountName string `json:"serviceAccountName,omitempty"`
```
Comment (nachocano) [inline pkg/apis/duck/v1/identity_types.go:36]: can you create a bug in the sources project to see whether we should remove this ServiceAccountName or change it to contain the GoogleServiceAccountName?

Comment (nachocano) [reply pkg/apis/duck/v1/identity_types.go:36]: Just did it so that we don't forget... https://github.com/google/knative-gcp/issues/1426

Comment (danyinggu) [reply pkg/apis/duck/v1/identity_types.go:36]: Thanks for creating the issue! Sorry for the delay to catch this comment.

**Pull request merged**